src/ui/display.c

```diff
@@ -305,10 +305,13 @@ bool prepare_display_output() {
     char b58address[B58_ADDRESS_LEN] = {0};
     uint8_t address[ADDRESS_LEN] = {0};
     if (address_from_pubkey_hash(output.pubkey_hash, PUBKEY_HASH_LEN, address, ADDRESS_LEN)) {
-        THROW(SW_INTERNAL_ERROR);
+        explicit_bzero(&G_context, sizeof(G_context));
+        io_send_sw(SW_INTERNAL_ERROR);
+        ui_menu_main();
+        return true;
     }
     base58_encode(address, ADDRESS_LEN, b58address, B58_ADDRESS_LEN);
-    memmove(g_address, b58address, sizeof(b58address));
+    memmove(g_address, b58address, B58_ADDRESS_LEN);
```

Comment:
Why are you changing the sizeof here?

Reply:
Just using the static value, b58address is actually `uint8_t[B58_ADDRESS_LEN]`.
This is what is used on the rest of the file, so I kept it like this.

Reply:
I'd rather keep the `sizeof(...)` to avoid checking the size of `b58address`.